UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -573,7 +592,7 @@ void SendInfoToBrain()
 
             // var param = m_PolicyFactory.brainParameters; // look, no brain params!
 
-            m_Info.reward = m_Reward;
+            m_Info.reward = rewardProvider.GetIncrementalReward();
```

Comment:
Would this be an appropriate place to record rewards during inference?

Reply:
I think Vince had proposed that we have a wrapper  IRewardProvider implementation for debugging.

Reply:
do we want to log cumulative reward, or incremental?

Reply:
I think incremental -  we'll probably still accumualte it, but that seems more fundamental.